poller/poller.go

```diff
@@ -65,19 +91,33 @@ func (poller *Poller) pollForChanges() {
 	}
 
 	// Update Cache
-	updatedCache := SplitData{
+	splitData := SplitData{
 		Splits:             splits,
 		Since:              since,
 		Segments:           segments,
 		UsingSegmentsCount: usingSegmentsCount,
 	}
+	serializedData, err := generateSerializedData(splitData)
+	if err != nil {
+		poller.Error <- err
+		return
+	}
+	updatedCache := Cache{
+		SplitData:      splitData,
+		SerializedData: serializedData,
+	}
 	atomic.StorePointer(&poller.cache, unsafe.Pointer(&updatedCache))
 
 }
 
-// GetSplitData returns cache results of SplitData
+// GetSplitData returns split data cache results
 func (poller *Poller) GetSplitData() SplitData {
```

Comment:
Do we use this anywhere?

Reply:
We do not. I wanted to keep the changes in this PR minimal. I can remove it in a subsequent PR